csharp/src/Ice/Acm.cs

```diff
@@ -43,50 +127,35 @@ internal class ConnectionFactoryAcmMonitor : IAcmMonitor
         private readonly Communicator _communicator;
         private readonly HashSet<Connection> _connections = new HashSet<Connection>();
         private readonly object _mutex = new object();
-        private List<Connection> _reapedConnections = new List<Connection>();
         private Timer? _timer;
 
         public void Add(Connection connection)
         {
-            if (Acm.Timeout == TimeSpan.Zero || Acm.Timeout == Timeout.InfiniteTimeSpan)
-            {
-                return;
-            }
-
-            lock (_mutex)
+            if (!Acm.IsDisabled)
             {
-                if (_connections.Count == 0)
+                lock (_mutex)
                 {
-                    _connections.Add(connection);
-                    _timer = new Timer(RunTimerTask, this, Acm.Timeout / 2, Acm.Timeout / 2);
-                }
-                else
-                {
-                    _changes.Add((connection, false));
+                    if (_connections.Count == 0)
```

Comment:
Although this is not new code, it would be nice to add a comment to explain what's happening here.

Reply:
Added the following comment:
```
                    // Unless it's the first connection, we don't add the connection directly to the _connections set.
                    // The _connections set can only be accessed by the timer when it's set. The goal here is to avoid
                    // copying _connections or holding a lock while iterating over all the connections to call Monitor.
                    // There could be thousands of connection in this set. So instead of modifying the connection set
                    // in Add/Remove, we add the Add/Remove request to the _changes list which is then processed by
                    // the timer to add/remove connections from the connection set.
```